Tests/MapboxCoreNavigationTests/PredictiveCache/PredictiveCacheManagerTests.swift

```diff
@@ -0,0 +1,93 @@
+        let expectedOptions = predictiveCacheOptions.predictiveCacheNavigationOptions.locationOptions
+        XCTAssertNil(navigator.passedTileStore)
+        XCTAssertNil(navigator.passedLocationTrackerOptions)
+        XCTAssertEqual(navigator.passedNavigationTrackerOptions?.currentLocationRadius, UInt32(expectedOptions.currentLocationRadius))
+        XCTAssertEqual(navigator.passedNavigationTrackerOptions?.destinationLocationRadius, UInt32(expectedOptions.destinationLocationRadius))
+        XCTAssertEqual(navigator.passedNavigationTrackerOptions?.routeBufferRadius, UInt32(expectedOptions.routeBufferRadius))
+
+        navigator.passedNavigationTrackerOptions = nil
+
+        manager.updateMapControllers(styleSourcePaths: ["first"])
+
+        XCTAssertNil(navigator.passedNavigationTrackerOptions, "Does not recreate navigation controller")
+    }
+
+    func testCreateNavigationAndMapsController() {
+        let manager = PredictiveCacheManager(predictiveCacheOptions: predictiveCacheOptions,
+                                             mapOptions: (tileStore: tilestore, styleSourcePaths: ["first"]),
+                                             navigator: navigator)
+        let expectedOptions = predictiveCacheOptions.predictiveCacheNavigationOptions.locationOptions
+        XCTAssertEqual(navigator.passedNavigationTrackerOptions?.currentLocationRadius, UInt32(expectedOptions.currentLocationRadius))
+        XCTAssertEqual(navigator.passedNavigationTrackerOptions?.destinationLocationRadius, UInt32(expectedOptions.destinationLocationRadius))
+        XCTAssertEqual(navigator.passedNavigationTrackerOptions?.routeBufferRadius, UInt32(expectedOptions.routeBufferRadius))
+
+        let expectedMapOptions = predictiveCacheOptions.predictiveCacheMapsOptions.locationOptions
+        XCTAssertTrue(navigator.passedTileStore == tilestore)
+        XCTAssertEqual(navigator.passedLocationTrackerOptions?.currentLocationRadius, UInt32(expectedMapOptions.currentLocationRadius))
+        XCTAssertEqual(navigator.passedLocationTrackerOptions?.destinationLocationRadius, UInt32(expectedMapOptions.destinationLocationRadius))
+        XCTAssertEqual(navigator.passedLocationTrackerOptions?.routeBufferRadius, UInt32(expectedMapOptions.routeBufferRadius))
+        XCTAssertEqual(navigator.passedCacheOptions?.maxAverageDownloadBytesPerSecond, 0)
+        XCTAssertEqual(navigator.passedCacheOptions?.dataset, "first")
+        XCTAssertEqual(navigator.passedCacheOptions?.dataDomain, .maps)
+        XCTAssertEqual(navigator.passedCacheOptions?.concurrency, predictiveCacheOptions.predictiveCacheMapsOptions.maximumConcurrentRequests)
+
+        navigator.passedLocationTrackerOptions = nil
+        navigator.passedNavigationTrackerOptions = nil
+
+        manager.updateMapControllers(styleSourcePaths: ["second"])
+
+        XCTAssertNil(navigator.passedNavigationTrackerOptions, "Does not recreate navigation controller")
```

Comment:
This basically repeats `testCreateNavigationController`

Reply:
I removed this line, but left checking of map cache controller creation